browser/infobars/request_otr_infobar_delegate.cc

```diff
@@ -63,11 +62,12 @@ std::u16string RequestOTRInfoBarDelegate::GetButtonLabel(
 bool RequestOTRInfoBarDelegate::Accept() {
   content::WebContents* web_contents =
       infobars::ContentInfoBarManager::WebContentsFromInfoBar(infobar());
-  RequestOTRStorageTabHelper* tab_storage =
-      RequestOTRStorageTabHelper::GetOrCreate(web_contents);
-  /* reload, not in Off-The-Record mode, and suppress interstitial */
-  tab_storage->set_is_proceeding(true);
-  tab_storage->set_requested_otr(false);
+  request_otr::RequestOTRService* request_otr_service =
+      request_otr::RequestOTRServiceFactory::GetForBrowserContext(
+          web_contents->GetBrowserContext());
+  DCHECK(request_otr_service);
+  request_otr_service->WithdrawOTR(web_contents->GetLastCommittedURL());
```

Comment:
I'm a little worried this could possibly be wrong in some edge cases if the page somehow navigated and the infobar is still shown? Not sure if that's possible, but seems safer to pass the url to the infobar when it gets created?

Reply:
pass URL to the infobar in https://github.com/brave/brave-core/commit/c83ae23838b8132c40d9ce007c338abf52804ae9